`IdentityProviderApi` cannot properly deserialize `/ipds` responses

### Describe the bug?

okta-sdk `v.14` that is using polymorphic deserialization  cannot correctly deserialize responses from  `/api/v1/idps/${idpId}` endpoint.

Below are a few tests to demonstrate the issue.
```java
    @Test
    void getIdentityProviderHasNullPolicy() {
        ApiClient apiClient = Clients.builder()
                .setOrgUrl("https://sso-development.<my-domain>.com")  
                .setClientCredentials(new TokenClientCredentials("<secret>"))
                .build();
        IdentityProviderApi identityProviderApi = new IdentityProviderApi(apiClient);
        IdentityProvider idp = identityProviderApi.getIdentityProvider("0oa13b4m2byJwpt420h8");
        assertNotNull(idp);
        assertNotNull(idp.getPolicy(), "idp.policy is NULL"); // the test fails here because 'policy' field could not be deserialized and Jackson was configured to ignore the error
    }
```
The same test but without ignoring the Jackson error
```java
    @Test
    void getIdentityProviderHasNullPolicy() {
        ApiClient apiClient = Clients.builder()
                .setOrgUrl("https://sso-development.<my-domain>.com")  
                .setClientCredentials(new TokenClientCredentials("<secret>"))
                .build();

        // override Jackson configuration set in ApiClient constructor
        apiClient.getObjectMapper().configure(DeserializationFeature.FAIL_ON_INVALID_SUBTYPE, true);
        
        IdentityProviderApi identityProviderApi = new IdentityProviderApi(apiClient);
        IdentityProvider idp = identityProviderApi.getIdentityProvider("0oa13b4m2byJwpt420h8");  // InvalidTypeIdException is thrown here
        assertNotNull(idp);
        assertNotNull(idp.getPolicy(), "idp.policy is NULL");
    }
```
Full stack trace from the test above
```
ApiException{code=0, responseHeaders=null, responseBody='null'}
	at com.okta.sdk.resource.client.ApiClient.invokeAPI(ApiClient.java:1099)
	at com.okta.sdk.resource.api.IdentityProviderApi.getIdentityProvider(IdentityProviderApi.java:908)
	at com.okta.sdk.resource.api.IdentityProviderApi.getIdentityProvider(IdentityProviderApi.java:858)
	at com.cvent.login.okta.api.IdentityProvidersApiClientTest.getIdentityProviderThrowsJacksonException(IdentityProvidersApiClientTest.java:197)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: com.fasterxml.jackson.databind.exc.InvalidTypeIdException: Could not resolve subtype of [simple type, class com.okta.sdk.resource.model.IdentityProviderPolicy]: missing type id property 'type' (for POJO property 'policy')
 at [Source: (String)"{"id":"0oa13b4m2byJwpt420h8","name":"Conference Cisco","status":"ACTIVE","created":"2021-11-26T06:29:51.000Z","lastUpdated":"2023-03-16T19:48:10.000Z","protocol":{"type":"SAML2","endpoints":{"sso":{"url":"https://cvent-conference-dev.onelogin.com/trust/saml2/http-post/sso/d93bfaec-0326-4f04-8993-5b7bfc46c585","binding":"HTTP-POST","destination":"https://cvent-conference-dev.onelogin.com/trust/saml2/http-post/sso/d93bfaec-0326-4f04-8993-5b7bfc46c585"},"acs":{"binding":"HTTP-POST","type":"INSTANCE"[truncated 1628 chars]; line: 1, column: 1557] (through reference chain: com.okta.sdk.resource.model.IdentityProvider["policy"])

	at com.fasterxml.jackson.databind.exc.InvalidTypeIdException.from(InvalidTypeIdException.java:43)
	at com.fasterxml.jackson.databind.DeserializationContext.missingTypeIdException(DeserializationContext.java:2094)
	at com.fasterxml.jackson.databind.DeserializationContext.handleMissingTypeId(DeserializationContext.java:1607)
	at com.fasterxml.jackson.databind.jsontype.impl.TypeDeserializerBase._handleMissingTypeId(TypeDeserializerBase.java:307)
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer._deserializeTypedUsingDefaultImpl(AsPropertyTypeDeserializer.java:211)
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer.deserializeTypedFromObject(AsPropertyTypeDeserializer.java:145)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeWithType(BeanDeserializerBase.java:1296)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:138)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:314)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4825)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3772)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3755)
	at com.okta.sdk.resource.client.ApiClient.deserialize(ApiClient.java:805)
	at com.okta.sdk.resource.client.ApiClient.processResponse(ApiClient.java:954)
	at com.okta.sdk.resource.client.ApiClient.invokeAPI(ApiClient.java:1072)
```

### What is expected to happen?

* Both unit tests above should pass
* `IdentityProviderApi` should return Identity providers with policy field properly populated from Okta `/idps/*` endpoint responses 

### What is the actual behavior?

`IdentityProviderApi#getIdentityProvider` returns an Identity Provider with NULL `policy` field even though the `policy` object is present in the REST response

### Reproduction Steps?

Please see the provided unit tests

### Additional Information?

_No response_

### Java Version

openjdk version "17.0.5" 2022-10-18 LTS
OpenJDK Runtime Environment Corretto-17.0.5.8.1 (build 17.0.5+8-LTS)
OpenJDK 64-Bit Server VM Corretto-17.0.5.8.1 (build 17.0.5+8-LTS, mixed mode, sharing)


### SDK Version

14.0.0

### OS version

Darwin C02G22CMMD6R 22.6.0 Darwin Kernel Version 22.6.0: Tue Nov  7 21:48:06 PST 2023; root:xnu-8796.141.3.702.9~2/RELEASE_X86_64 x86_64

--- a/api/pom.xml
+++ b/api/pom.xml
@@ -21,7 +21,7 @@
     <parent>
         <groupId>com.okta.sdk</groupId>
         <artifactId>okta-sdk-root</artifactId>
-        <version>14.1.0-SNAPSHOT</version>
+        <version>15.0.0-SNAPSHOT</version>
     </parent>
 
     <artifactId>okta-sdk-api</artifactId>

--- a/coverage/pom.xml
+++ b/coverage/pom.xml
@@ -21,7 +21,7 @@
     <parent>
         <groupId>com.okta.sdk</groupId>
         <artifactId>okta-sdk-root</artifactId>
-        <version>14.1.0-SNAPSHOT</version>
+        <version>15.0.0-SNAPSHOT</version>
     </parent>
 
     <artifactId>okta-sdk-coverage</artifactId>

--- a/examples/pom.xml
+++ b/examples/pom.xml
@@ -21,7 +21,7 @@
     <parent>
         <groupId>com.okta.sdk</groupId>
         <artifactId>okta-sdk-root</artifactId>
-        <version>14.1.0-SNAPSHOT</version>
+        <version>15.0.0-SNAPSHOT</version>
     </parent>
 
     <artifactId>okta-sdk-examples</artifactId>

--- a/examples/quickstart/pom.xml
+++ b/examples/quickstart/pom.xml
@@ -20,7 +20,7 @@
     <parent>
         <groupId>com.okta.sdk</groupId>
         <artifactId>okta-sdk-examples</artifactId>
-        <version>14.1.0-SNAPSHOT</version>
+        <version>15.0.0-SNAPSHOT</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/impl/pom.xml
+++ b/impl/pom.xml
@@ -21,7 +21,7 @@
     <parent>
         <groupId>com.okta.sdk</groupId>
         <artifactId>okta-sdk-root</artifactId>
-        <version>14.1.0-SNAPSHOT</version>
+        <version>15.0.0-SNAPSHOT</version>
     </parent>
 
     <artifactId>okta-sdk-impl</artifactId>

--- a/integration-tests/pom.xml
+++ b/integration-tests/pom.xml
@@ -20,7 +20,7 @@
     <parent>
         <groupId>com.okta.sdk</groupId>
         <artifactId>okta-sdk-root</artifactId>
-        <version>14.1.0-SNAPSHOT</version>
+        <version>15.0.0-SNAPSHOT</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
 

--- a/pom.xml
+++ b/pom.xml
@@ -25,7 +25,7 @@
 
     <groupId>com.okta.sdk</groupId>
     <artifactId>okta-sdk-root</artifactId>
-    <version>14.1.0-SNAPSHOT</version>
+    <version>15.0.0-SNAPSHOT</version>
     <packaging>pom</packaging>
 
     <name>Okta Java SDK</name>
@@ -68,12 +68,12 @@
             <dependency>
                 <groupId>com.okta.sdk</groupId>
                 <artifactId>okta-sdk-api</artifactId>
-                <version>14.1.0-SNAPSHOT</version>
+                <version>15.0.0-SNAPSHOT</version>
             </dependency>
             <dependency>
                 <groupId>com.okta.sdk</groupId>
                 <artifactId>okta-sdk-impl</artifactId>
-                <version>14.1.0-SNAPSHOT</version>
+                <version>15.0.0-SNAPSHOT</version>
             </dependency>
 
             <!-- Other Okta Projects -->
@@ -103,14 +103,14 @@
             <dependency>
                 <groupId>com.okta.sdk</groupId>
                 <artifactId>okta-sdk-integration-tests</artifactId>
-                <version>14.1.0-SNAPSHOT</version>
+                <version>15.0.0-SNAPSHOT</version>
             </dependency>
 
             <!-- Examples -->
             <dependency>
                 <groupId>com.okta.sdk</groupId>
                 <artifactId>okta-sdk-examples-quickstart</artifactId>
-                <version>14.1.0-SNAPSHOT</version>
+                <version>15.0.0-SNAPSHOT</version>
             </dependency>
 
             <!-- Logging -->

--- a/src/swagger/api.yaml
+++ b/src/swagger/api.yaml
@@ -26122,27 +26122,25 @@ components:
         - DELTA_CRL
         - OCSP
     IdentityProviderPolicy:
-      allOf:
-        - $ref: '#/components/schemas/Policy'
-        - type: object
-          properties:
-            accountLink:
-              $ref: '#/components/schemas/PolicyAccountLink'
-            conditions:
-              $ref: '#/components/schemas/PolicyRuleConditions'
-            mapAMRClaims:
-              type: boolean
-              description: Enable mapping AMR from IdP to Okta to downstream apps
-              default: false
-              x-okta-lifecycle:
-                features:
-                  - IDP_AMR_CLAIMS_MAPPING
-            maxClockSkew:
-              type: integer
-            provisioning:
-              $ref: '#/components/schemas/Provisioning'
-            subject:
-              $ref: '#/components/schemas/PolicySubject'
+      type: object
+      properties:
+        accountLink:
+          $ref: '#/components/schemas/PolicyAccountLink'
+        conditions:
+          $ref: '#/components/schemas/PolicyRuleConditions'
+        mapAMRClaims:
+          type: boolean
+          description: Enable mapping AMR from IdP to Okta to downstream apps
+          default: false
+          x-okta-lifecycle:
+            features:
+              - IDP_AMR_CLAIMS_MAPPING
+        maxClockSkew:
+          type: integer
+        provisioning:
+          $ref: '#/components/schemas/Provisioning'
+        subject:
+          $ref: '#/components/schemas/PolicySubject'
     IdentityProviderPolicyProvider:
       type: string
       enum:
@@ -28508,7 +28506,7 @@ components:
         mapping:
           ACCESS_POLICY: '#/components/schemas/AccessPolicy'
           OAUTH_AUTHORIZATION_POLICY: '#/components/schemas/AuthorizationServerPolicy'
-          IDP_DISCOVERY: '#/components/schemas/IdentityProviderPolicy'
+          IDP_DISCOVERY: '#/components/schemas/AccessPolicy'
           MFA_ENROLL: '#/components/schemas/MultifactorEnrollmentPolicy'
           OKTA_SIGN_ON: '#/components/schemas/OktaSignOnPolicy'
           PASSWORD: '#/components/schemas/PasswordPolicy'

